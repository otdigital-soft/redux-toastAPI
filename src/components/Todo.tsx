import React from 'react';

import ListItem from '@material-ui/core/ListItem';
import ListItemSecondaryAction from '@material-ui/core/ListItemSecondaryAction';
import ListItemText from '@material-ui/core/ListItemText';
import IconButton from '@material-ui/core/IconButton';
import DeleteIcon from '@material-ui/icons/Delete';
import { ITodos } from '../types';
 
const Todo: React.FC<ITodos> = ({ text, id, clickDelete }) => {

  return (
    <ListItem key={id}  ContainerComponent="div">
      <ListItemText primary={text} />
      <ListItemSecondaryAction>
        <IconButton
          aria-label="Delete"
          onClick={() => {
            clickDelete(id);
          }}
        >
          <DeleteIcon />
        </IconButton>
      </ListItemSecondaryAction>
</ListItem>
  )
  };

export default Todo;
