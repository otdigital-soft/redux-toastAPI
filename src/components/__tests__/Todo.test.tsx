import React from 'react';
import renderer from 'react-test-renderer';
import Todo from '../Todo';

describe('Test Todo Component', () => {
    it('should render without error', () => {
    const props = {
        text: 'title',
        _id: 'Author',
        clickDelete: () => {}
    }
    const component = renderer.create(<Todo {...props} />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
    });
});
