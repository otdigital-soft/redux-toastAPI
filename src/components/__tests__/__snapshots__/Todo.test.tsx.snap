// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Todo Component should render without error 1`] = `
<div
  className="MuiListItem-container"
>
  <div
    className="MuiListItem-root MuiListItem-gutters MuiListItem-secondaryAction"
    disabled={false}
  >
    <div
      className="MuiListItemText-root"
    >
      <span
        className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
      >
        title
      </span>
    </div>
  </div>
  <div
    className="MuiListItemSecondaryAction-root"
  >
    <button
      aria-label="Delete"
      className="MuiButtonBase-root MuiIconButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiIconButton-label"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
          />
        </svg>
      </span>
    </button>
  </div>
</div>
`;
